### PR TITLE
Remove device override for operator new when the C++ standard >= 26

### PR DESCRIPTION
**Pull request by OgnianM**
Related to https://github.com/llvm/llvm-project/issues/114048

### CODE REVIEW

File: clang/lib/Headers/cuda_wrappers/new [hunk 1]
```diff
@@ -91,12 +91,14 @@ __device__ inline void operator delete[](void *ptr,
 #endif
 
 // Device overrides for placement new and delete.
+#if _LIBCPP_STD_VER < 26
```
Comment (Artem-B) [inline clang/lib/Headers/cuda_wrappers/new:94]: This helps with libc++. 

Do we run into this issue with libstdc++, too? If so, we may check its version here, too.

Comment (OgnianM) [reply clang/lib/Headers/cuda_wrappers/new:94]: Looks like it
https://github.com/gcc-mirror/gcc/blob/9dd9a88b75334bc079b8ab5fb2dbb5d56765bd60/libstdc%2B%2B-v3/libsupc%2B%2B/new#L198-L214

I have an older version of libstdc++ on my system, so I can't easily test it, but it looks like changing this to
`#if _LIBCPP_STD_VER < 26 || __cpp_lib_constexpr_new < 202406L` should work.

Comment (Artem-B) [reply clang/lib/Headers/cuda_wrappers/new:94]: Yup. Reproducible on compiler explorer: https://godbolt.org/z/xr5nEhGnr

Comment (OgnianM) [reply clang/lib/Headers/cuda_wrappers/new]: Alright, both should be fixed now.

**Review by Artem-B — APPROVED** [review]

**Comment by OgnianM** [comment]
Ping

**Comment by Artem-B** [comment]
The patch request has been already approved. Nothing seems to have changed since then to the patch?
Are you asking for help landing the patch?

**Comment by OgnianM** [comment]
Yeah, am I supposed to do anything at this point or should I just leave it to you?

**Pull request merged**